toolchain/sem_ir/entity_with_params_base.h

```diff
@@ -95,21 +94,27 @@ struct EntityWithParamsBase {
   // A block containing the pattern insts for the parameter lists.
   InstBlockId pattern_block_id;
   // A block containing, for each implicit parameter, a reference to the
-  // instruction in the entity's declaration block that depends on all other
-  // pattern-match insts pertaining to that parameter.
-  InstBlockId implicit_param_refs_id;
-  // A block containing, for each implicit parameter, a reference to the
   // instruction in the entity's pattern block that depends on all other
   // pattern insts pertaining to that parameter.
   InstBlockId implicit_param_patterns_id;
   // A block containing, for each explicit parameter, a reference to the
-  // instruction in the entity's declaration block that depends on all other
-  // pattern-match insts pertaining to that parameter.
-  InstBlockId param_refs_id;
-  // A block containing, for each explicit parameter, a reference to the
   // instruction in the entity's pattern block that depends on all other
   // pattern insts pertaining to that parameter.
   InstBlockId param_patterns_id;
+  // If this entity is a function, this block consists of references to the
+  // `AnyParam` insts that represent the function's call parameters. The "call
+  // parameters" are the parameters corresponding to the arguments that are
+  // passed to a `Call` inst, so they include `self` (if applicable) and the
```

Comment:
Why does this say "call parameters" instead of "`Call`"? Had you considered something like:

```suggestion
  // `AnyParam` insts that represent the function's `Call` parameters. They include `self` (if applicable) and the
```

Reply:
I want to be a little more explicit about the fact that I'm defining this as a term of art, particularly so that it can be reused in places like `name_component.h`. It hadn't occurred to me to say "`Call`"  instead of "call", but I suppose that works; done.

Reply:
FWIW, I think in general we should be able to say "instruction X parameters" and have it be clear what it means without calling it a term of art.

Reply:
> Isn't this just saying that the `Call` parameters are parameters to the `Call` instruction?

I wouldn't put it that way, no. `EntityWithParamsBase` represents part of the declaration of a callable entity; it doesn't represent a particular call to that entity, so there's no `Call` inst in "scope" here. By the same token, `Call` doesn't have parameters, it has arguments. Does that make sense?

> Although, this also raises another question. Maybe this comment should be on `Call` so that when we say "`Call` parameters" the answer is found on `Call` without needing to say "look at `EntityWithParamsBase`"? Sorry about this, since I think this has already moved around a little.

That could work, but I think it would be awkward, for the same reason: the `Call` inst represents a particular call to a callable entity, so I think it would feel like a non sequitur to talk about how that callable entity represents its parameters in its own IR.

(While looking at this I noticed that the `Call` documentation is out of date, so I fixed that.)

Reply:
> > Isn't this just saying that the `Call` parameters are parameters to the `Call` instruction?
> 
> I wouldn't put it that way, no. `EntityWithParamsBase` represents part of the declaration of a callable entity; it doesn't represent a particular call to that entity, so there's no `Call` inst in "scope" here. By the same token, `Call` doesn't have parameters, it has arguments. Does that make sense?

FWIW, here's a different phrasing that I don't know if it'd work for you:

```
`Call` parameters correspond to `Call::args_id`, but in parameter form.
```

I'll leave it to you whether to adopt something like that.

> > Although, this also raises another question. Maybe this comment should be on `Call` so that when we say "`Call` parameters" the answer is found on `Call` without needing to say "look at `EntityWithParamsBase`"? Sorry about this, since I think this has already moved around a little.
> 
> That could work, but I think it would be awkward, for the same reason: the `Call` inst represents a particular call to a callable entity, so I think it would feel like a non sequitur to talk about how that callable entity represents its parameters in its own IR.
> 
> (While looking at this I noticed that the `Call` documentation is out of date, so I fixed that.)

To be clear here, what I really mean is instead of having things like "(see entity_with_params_base.h for the definition of that term)." in `pattern_match.h`, maybe having it on `Call` would mean that people seeing "`Call` parameters" and wondering what that means would be able to find it on `Call` without the explicit reference. I'm not meaning to argue the detail in this half of the comment.

Reply:
> FWIW, here's a different phrasing that I don't know if it'd work for you:
> 
> ```
> `Call` parameters correspond to `Call::args_id`, but in parameter form.
> ```
> 
> I'll leave it to you whether to adopt something like that.

I'm afraid the "in parameter form" part doesn't work for me -- they don't just have different forms, they have different content, because they represent categorically different entities.

That said, I don't want to just shoot down your suggestions; this concept is kind of subtle, so it's important for the explanation to be clear, and evidently it isn't. If you have any thoughts about how to fix it, I'm happy to iterate more in a follow-up PR.